docs/dev_setup.rst

```diff
@@ -11,12 +11,58 @@ This section describes how someone wanting to contribute to the libCellML projec
+* Windows *git* is available from a variety of vendors we commonly use `git for windows <http://gitforwindows.org/>`_, but other popular *git* implementations are:
+
+  - `GitHub Desktop <https://desktop.github.com/>`_
+  - `GitKracken <https://www.gitkraken.com/>`_
+  - `Git SCM <https://git-scm.com/>`_
+
+  The choice of implementation is one of personal preference.
+
+CMake
+-----
+
+*CMake* is the cross-platform family of tools designed to build, test and package software.  *CMake* is used to control the software compilation process using simple platform and compiler independent configuration files, and generate native makefiles and workspaces that can be used in the compiler environment of your choice.
+
+Again installation of *CMake* is particular to each operating system.  For Ubuntu (and other linux distributions) *CMake* can be installed using the package manager with the command :code:`sudo apt-get install cmake`.  For macOS and Windows *CMake* provides installation binaries from `here <https://cmake.org/download/>`_ choose the binary appropriate for your operating system and follow the installation instructions.
+
+Toolchain
+---------
+
+The toolchain specifies the compiler that we will use to build *libCellML*, toolchains are highly dependent on the operating system.  When we test *libCellML* we currenly use the *clang* family of compilers on macOS, the *GNU* family of compilers on Ubuntu, and *Microsoft Visual Studio Compilers* on Windows.  We recommend using these compilers on these systems, but feel free to use a different toolchain.  We often use *intel* compilers to build *libCellML* but we don't (at the time of writing) test with them.
+
+The following sub-sections below provide guidance on how to install the recommended toolchain on the major operating systems that *libCellML* supports.
+
+Windows
++++++++
+
+*Microsoft Visual Studio* is available to download from `here <https://www.visualstudio.com/downloads/>`_.  We currently test with version 14 of this software but later versions are known to work.  The *community edition* is more than sufficient for the needs of *libCellML*.  To minimize the size of the installation you may install only the C++ compiler, this component (and it's requirements) is sufficient for building the *libCellML* library.
+
+Setting up the codebase
+=======================
+
+The remainder of this document assumes that the above pre-requisites have been met.  This document covers setup from the command line if you are using a GUI like 'GitHub for Windows' then you will need to adjust the commands for the GUI you are using.
 
 The goal here is to get a working copy of source code, tests, and documentation onto your PC so that you can begin development.  To make this happen you will need to fork the :term:`prime libCellML repository`, make a clone onto your PC, and set up the Git remotes.  In :numref:`fig_devSetup_githubRepos` you can see a pictorial representation of what we are aiming to achieve.
```

Comment:
The goal here is to get a working copy of source code, tests, and documentation onto your computer so that you can begin development.  To make this happen, you will need to fork the :term:`prime libCellML repository`, make a clone onto your computer, and set up the Git remotes (see below).  In :numref:`fig_devSetup_githubRepos`, you can see a pictorial representation of what we are aiming to achieve.

Note: might decide between "Git" and "git". It's not consistent at the moment.